### ISSUE TITLE
Strange errors in IntelliJ (Possible collision / interference with Lombok)

### ISSUE THREAD

**Issue by johnmcclean** [derive4j/hkt#30]
Hey guys,

I'd like to turn on this feature in cyclops-react (https://github.com/aol/cyclops-react/issues/347), when I added the hkt project as a dependency and make cyclops-react Higher class extend org.derive4j.hkt.__ everything compiles just fine from the command line. Unfortunately, IntelliJ starts showing some strange errors instead. 

Lombok's val will generally fail, but other errors seem unrelated and really strange. I've attached a screenshot where the line numbers of some spurious compile errors have been misattributed.

I'm not sure how easy or hard it would be to fix this (as the issue may well be on the IntelliJ side or Lomboks), but I thought you'd appreciate getting a report of it.

Thanks,
John

<img width="1368" alt="screen shot 2017-05-16 at 22 02 24" src="https://cloud.githubusercontent.com/assets/9964792/26128239/ad7bdc76-3a83-11e7-8e1c-140edc4a10e0.png">

**Comment by jbgi** [comment]
Can you try `master` on this repo on https://github.com/jbgi/cyclops-react and do a ` ./gradlew compileJmhJava compileTestJava` then a "Refresh all gradle projects" in Idea then finally "Rebuild Project".
It kind-of workd for me under intellij (2017.1.1) although not sure the generated java files are correctly setup in a source directory by intellij.

**Comment by johnmcclean** [comment]
Yeah that works, IntelliJ will pick up the class files generated via the Gradle build, and oddly enough you can even change those classes one at a time and incremental compilation seems to work. Doing a rebuild in IntelliJ causes the problem to re-emerge though.

**Comment by jbgi** [comment]
@johnmcclean-aol after that I also did a `./gradlew clean` and a "Rebuild Project" via intellij and it worked. Strange this is not working for you.

**Comment by johnmcclean** [comment]
Yeah that would also work, what I mean though is it doesn't help IntelliJ do the compilation itself. It can read the class files generated by javac called via Gradle.

**Comment by jbgi** [comment]
but after a "./gradlew clean" there should be no "class files generated by javac called via Gradle" left, no? will check in more details tonight though.

**Comment by johnmcclean** [comment]
Ah, sorry - I see what you mean now. I was running clean compile. Let me give that a try..

**Comment by mchiareli** [comment]
that may be related to the order processors will run? if so some changes are needed to have it working with lombok:
https://github.com/rzwitserloot/lombok/issues/973
